source/common/modules/markdown-editor/util/strip-duplicate-spaces.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        stripDuplicateSpaces function
+ * CVM-Role:        Utility function
+ * Maintainers:     Rich Douglas
+ * License:         GNU GPL v3
+ *
+ * Description:     This function strips duplicate spaces from a selected region
+ *                  of text.
+ *
+ * END HEADER
+ */
+
+import { ChangeSet, StateCommand } from '@codemirror/state'
+
+/**
+ * Utility to strip duplicate spaces from a selected region of text.
+ */
+export const stripDuplicateSpaces: StateCommand = ({ state, dispatch }) => {
```

Comment:
Changed this from a `Command` -- accepting `EditorView` -- to a `StateCommand` -- which just requires `state` and `dispatch`.

* [StateCommand](https://codemirror.net/docs/ref/#state.StateCommand)

This makes it easy to unit test the command 'cos we can use doubles (stubs/mocks) to assert whether the command is making th desired changes to the editor's state.